wandb/util.py

```diff
@@ -1477,3 +1477,18 @@ def check_dict_contains_nested_artifact(d, nested=False):
         ) and nested:
             return True
     return False
+
+
+def deprecate(field_name: str, warning_message: str):
+    """
+    Warn the user that a feature has been deprecated
+    and store the information about the event in telemetry.
+
+    Args:
+        field_name: The name of the feature that has been deprecated.
+                    Defined in wandb/proto/wandb_telemetry.proto::Deprecated
+        warning_message: The message to display to the user.
+    """
+    with wandb.wandb_lib.telemetry.context(run=wandb.run) as tel:
+        setattr(tel.deprecated, field_name, True)
```

Comment:
using strings and setattr seems error prone, thats why the context manager was nice by typing the proto record

Reply:
How about https://github.com/wandb/client/blob/778ee9e0d8625ec70e5f63f436d4347204bcd768/wandb/sdk/lib/deprecate.py
?

Reply:
it is better, but it means that mistakes are caught by a runtime check rather than a static checker, which means we have to have full code coverage to avoid issues   This is why i chose the awkward context manager method with telemetry. 
other choices that might be less awkward to handle this would be to create a deprecation class based on the protobuf during the `make proto` stage?

Reply:
or a generated enum maybe?   just some thoughts.